### PR TITLE
Updated service prototype with new API framework.

### LINKED ISSUE #55: Figure out a clever way to autogenerate openapi.yaml
The file `openapi.yaml`, which is used to generate our static HTML REST API docs, is not particularly easy to write or navigate. And the Credit Engine schema's documentation is embedded in it, too, which introduces a source of volatility. It might be good for us to ponder ways of assembling this file from different pieces. A simple manually-run Python script might be good enough to get us what we want.

## Interesting Things

* [Huma](https://huma.rocks/) - A modern, simple, fast & flexible micro framework for building HTTP REST/RPC APIs in Golang backed by OpenAPI 3 and JSON Schema

### PR DESCRIPTION
**Pull request by jeff-cohere**
This is my first stab at using [Huma](https://huma.rocks/) to implement the endpoints for the DTS. Huma frames things more explicitly in terms of input (for requests) and output (for responses) in a way that allows the definition and use of structs annotated with tags. These tags can include documentation (with examples), and remove a lot of marshalling/unmarshalling boilerplate code. I quite like how it feels so far, but I think it needs another pair of eyes before we adopt it.

The motivation for this PR wasn't actually all this cleanup and additional explication. It turns out that Huma can take the struct annotations for the various endpoints and requests and turn them into a fully OpenAPI-compliant documentation site! I was doing this with a manually written `openapi.yaml` file (_blecch!_) and some file-embedding trickery, and this is definitely much nicer.

I'm planning on testing this in NERSC Spin. But I wanted to get this up before going too much further.

Closes #55

### CODE REVIEW

**Comment by jeff-cohere** [comment]
I've verified that this updated version of the DTS works properly in Spin, just like the old one.

File: services/transfer_service.go [hunk 1]
```diff
@@ -40,32 +48,38 @@ func writeError(w http.ResponseWriter, message string, code int) {
...
 }
 
 // a request for a file transfer (POST)
 type TransferRequest struct {
 	// name of source database
-	Source string `json:"source"`
+	Source string `json:"source" example:"jdp" doc:"source database identifier"`
 	// identifiers for files to be transferred
-	FileIds []string `json:"file_ids"`
+	FileIds []string `json:"file_ids" doc:"source-specific identifiers for files to be transferred"`
```
Comment (ialarmedalien) [inline services/transfer_service.go:74]: is it possible to add `example`s for data structures like an array?

Comment (jeff-cohere) [reply services/transfer_service.go:74]: I think we could do this with a tag like `example:"[\"fileid1\", \"fileid2\"]"`, but I haven't tried it.

Comment (ialarmedalien) [reply services/transfer_service.go:74]: Just reading the documentation for Huma and it looks like there are all kinds of fancy things possible -- as long as you believe!

File: services/prototype.go [hunk 2]
```diff
@@ -81,261 +77,179 @@ func getAuthInfo(header http.Header) (string, string, error) {
 			orcid = orcids[0]
 		}
 	}
-	return accessToken, orcid, err
+	if err != nil {
+		return orcid, huma.Error401Unauthorized(err.Error())
+	} else {
+		return orcid, nil
+	}
```
Comment (ialarmedalien) [inline services/prototype.go:84]: don't really need the `else` after a `return`

File: services/transfer_service.go (hunk 1 above)
Comment (jeff-cohere) [reply services/transfer_service.go:74]: I'd love to know what you think of Huma, though I know you're new to Golang. I distrust magical things that do everything for you, but Huma seems to hike right up to where it might start violating the Principle of Least Astonishment and then stops. :-)

**Review by ialarmedalien — APPROVED** [review]

File: services/transfer_service.go (hunk 1 above)
Comment (ialarmedalien) [reply services/transfer_service.go:74]: It looked pretty nice from the docs--similar to a couple of other frameworks I've used--and I like the JSONschema integration. You need some nice edge cases to test it with, though -- it's easy for the docs to talk up all the fancy features but neglect to mention that there's no support for <critical feature x> or anything that's slightly outside what the original dev had imagined.

Comment (jeff-cohere) [reply services/transfer_service.go:74]: I'll try to stick in some more complicated doc examples like arrays and objects before we merge this. Let me know if there's anything specific you'd like me to try.

Comment (jeff-cohere) [reply services/transfer_service.go:74]: The array example you mentioned does work as expected. Objects seem to be handled by field doc tags. So for now I'd say we're probably safe going with Huma. I'm sure we'll encounter some interesting edge cases as we flesh things out!
